xdsl/printer.py

```diff
@@ -641,13 +641,17 @@ def _print_attr_string(self, attr_tuple: tuple[str, Attribute]) -> None:
             self.print(f'"{attr_tuple[0]}" = ')
             self.print_attribute(attr_tuple[1])
 
-    def print_op_attributes(self, attributes: dict[str, Attribute]) -> None:
-        if len(attributes) == 0:
-            return
+    def print_op_attributes(
+        self, attributes: dict[str, Attribute], reserved_attr_names: Iterable[str] = []
+    ) -> None:
+        attribute_list = list(
```

Comment:
```suggestion
        attribute_list = [
```

Reply:
OOC; is it really different in that case? Aren't both of them calling the list constructor on the same Generator?
No feeling about it, applying your change, but wondering